### ISSUE TITLE
Inconsistent speed in rapid

### ISSUE THREAD

**Issue by irvall** [niklasf/lila-gif#4]
First of all, thanks for your contribution to lichess, and I _love_ the use of Rust.

I just played a fun attacking game as white that I wanted to share to my **all** my friends, but most of the opening moves, and also towards the end, is played back way too fast to follow. It could seem like you use some kind of algorithm that utilizes the entire game length as a variable – and thus could create funky results with fast/premoves in longer time controls (this was rapid 10+0). I will go read the source now.

**TLDR: Fast/pre-moves in rapid games are replayed way too fast in GIF**

**Comment by niklasf** [comment]
Could you please add the GIF as an example, just so we're talking about the same thing?

The decision about the frame lengths is made here: https://github.com/ornicar/lila/blob/2d3346c28c4e12dd4c16d81d48c61e16c9c1915b/modules/game/src/main/GifExport.scala#L88-L94

It's then sent to this Rust service, which renders it with whichever frame length's lila decided.

**Comment by irvall** [comment]
Ah, very sorry – here you go: https://lichess1.org/game/export/gif/white/2wp911hP.gif

The move early on was played fairly fast, but no near what's replayed in the GIF (I am no Penguin).

**Comment by niklasf** [comment]
Ok, yeah, that's extreme. Patches or suggestions for other schaling schemes welcome. Ideally it would still be close to real-time for bullet games.
